### ISSUE TITLE
[ISSUE] Issue with Unity Catalog resources that are on Workspace API

### ISSUE THREAD

**Issue by zshao9** [databricks/terraform-provider-databricks#3018]
I have a list of Unity Catalogs (one for each region) whose catalog/schemas/grants need to be created with Terraform.

Since those resources need to go to the Workspace-API providers, I need a list of providers one for each workspace (associated with the Unity Catalog).

However, Terraform doesn't support a dynamic list of providers (for for_each).  See:
* 2019 issue: https://github.com/hashicorp/terraform/issues/19932
* 2020 issue https://github.com/hashicorp/terraform/issues/24476
* 2022 discussions: https://stackoverflow.com/questions/72541598/using-for-each-with-provider-setting

Can we consider to use "metastore_id" properly on the catalog/schema/grants resources?
Then I just need to use a single provider with whatever workspace for all the different metastores.

**Comment by alexott** [comment]
No, it's not possible because such functionality isn't available in the UC APIs.

**Comment by mgyucht** [comment]
Just to expand on this a bit: workspace-level UC APIs only allow interacting with the metastore assigned to that workspace. Even if the TF provider were to accept a metastore ID, it wouldn't be able to interact with that metastore without a running workspace that is assigned to said metastore. We're working with the UC team to align on exposing these APIs at the account-level, in which case you would be able to interact with them purely when configured at the account level.

**Comment by bobdoah** [comment]
This would be incredibly useful

**Comment by sreedhanyagn** [comment]
We have more than 100 workspaces our organization and defining provider for each workspace is not feasible. Is there any alternative solution for this ? 
Can we use a single workspace as provider and create Catalogs for other workspaces using this provider ?

**Comment by zshao9** [comment]
As long as they share the same unity catalog (e.g. in the same region) then
this is possible.

Curious what is the use case of 100+ workspaces?  One per user?

**Comment by sreedhanyagn** [comment]
Hey @zshao9 , thanks for your reply. 
We are supporting many teams and each has team has multiple workspaces in different environments.

**Comment by zshao9** [comment]
Got it.  Are they all in the same region?  If so, then they use the same
unity catalog.  You can just set up an "admin workspace" where all these
Terraform operations are done.

**Comment by sreedhanyagn** [comment]
Yes, they all are in the same region.
